ingestion/src/metadata/ingestion/source/mlmodel/sagemaker.py

```diff
@@ -0,0 +1,176 @@
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+"""SageMaker source module"""
+
+import traceback
+from typing import Iterable, List, Optional
+
+from pydantic import BaseModel, Extra, Field, ValidationError
+
+from metadata.generated.schema.api.data.createMlModel import CreateMlModelRequest
+from metadata.generated.schema.entity.data.mlmodel import (
+    MlFeature,
+    MlHyperParameter,
+    MlStore,
+)
+from metadata.generated.schema.entity.services.connections.metadata.openMetadataConnection import (
+    OpenMetadataConnection,
+)
+from metadata.generated.schema.entity.services.connections.mlmodel.sageMakerConnection import (
+    SageMakerConnection,
+)
+from metadata.generated.schema.metadataIngestion.workflow import (
+    Source as WorkflowSource,
+)
+from metadata.generated.schema.type.entityReference import EntityReference
+from metadata.generated.schema.type.tagLabel import TagLabel
+from metadata.ingestion.api.source import InvalidSourceException
+from metadata.ingestion.source.mlmodel.mlmodel_service import MlModelServiceSource
+from metadata.utils.filters import filter_by_mlmodel
+from metadata.utils.logger import ingestion_logger
+
+logger = ingestion_logger()
+
+
+class SageMakerModel(BaseModel):
+    class Config:
+        extra = Extra.forbid
+
+    name: str = Field(..., description="Model name", title="Model Name")
+    arn: str = Field(..., description="Model ARN in AWS account", title="Model ARN")
+    creation_timestamp: str = Field(
+        ...,
+        description="Timestamp of model creation in ISO format",
+        title="Creation Timestamp",
+    )
+
+
+class SagemakerSource(MlModelServiceSource):
+    """
+    Source implementation to ingest SageMaker data.
+
+    We will iterate on the ML Models
+    and prepare an iterator of CreateMlModelRequest
+    """
+
+    def __init__(self, config: WorkflowSource, metadata_config: OpenMetadataConnection):
+        super().__init__(config, metadata_config)
+        self.sagemaker = self.connection.client
+
+    @classmethod
+    def create(cls, config_dict, metadata_config: OpenMetadataConnection):
+        config: WorkflowSource = WorkflowSource.parse_obj(config_dict)
+        connection: SageMakerConnection = config.serviceConnection.__root__.config
+        if not isinstance(connection, SageMakerConnection):
+            raise InvalidSourceException(
+                f"Expected SageMakerConnection, but got {connection}"
+            )
+        return cls(config, metadata_config)
+
+    def get_mlmodels(  # pylint: disable=arguments-differ
+        self,
+    ) -> Iterable[SageMakerModel]:
+        """
+        List and filters models
+        """
+        for model in self.sagemaker.list_models()["Models"]:
```

Comment:
could we please add some exception management in case:
- there is an issue picking up the models?
- There is a pydantic ValidationError when creating the model?
The goal of a workflow is that it should be as "safe" as possible, so even if we have an issue with a single model, the pipeline itself should keep running and processing further models